compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -1115,15 +1121,25 @@ public void visit(BLangFail failNode) {
 
     @Override
     public void visit(BLangSimpleVariableDef astVarDefStmt) {
+        VarKind kind;
+        if (astVarDefStmt.var.symbol.origin == SymbolOrigin.VIRTUAL) {
+            kind = VarKind.SYNTHETIC;
+        } else {
+            kind = VarKind.LOCAL;
+        }
```

Comment:
Do we have to do this change? What if we left is as LOCAL ? @dulvinw @riyafa

Reply:
I need this varkind to stop generating localvariables at the codegen phase.

Reply:
ok. so @riyafa, if we update the liveness analysis phase to look for both SYNTEHTIC and LOCAL vars, will that solve your issue?

Btw, hows the BIR optimization test cases are passing?

Reply:
Because the test cases have been modified. Please create an issue to fix this

Reply:
Maybe there are other places that rely on these checks. We might need to make sure we create issues for those cases as well.